src/EventGrid/EventGrid/SystemTopic/GetAzureEventGridSystemTopic.cs

```diff
@@ -0,0 +1,177 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.EventGrid.Models;
+using Microsoft.Azure.Commands.EventGrid.Utilities;
+using Microsoft.Azure.Management.EventGrid.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.EventGrid
+{
+    /// <summary>
+    /// 'Get-AzEventGridSystemTopic' Cmdlet gives the details of a / List of EventGrid topic(s)
+    /// <para> If Topic name provided, a single Topic details will be returned</para>
+    /// <para> If Topic name not provided, list of Topics will be returned</para>
+    /// </summary>
+
+    [Cmdlet(
+        "Get",
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "EventGridSystemTopic",
+        DefaultParameterSetName = ResourceGroupNameParameterSet),
+    OutputType(typeof(PSSystemTopic), typeof(PSSytemTopicListInstance))]
+
+    public class GetAzureRmEventGridSystemTopic : AzureEventGridCmdletBase
+    {
+        [Parameter(
+            Mandatory = false,
```

Comment:
Why all parameters are optional? As we discussed before, in certain parameter set, some parameters should be required. Please confirm that

Reply:
No, this is for getting system topics for a given subscription id. SubscriptionId is doesn't need to be separately passed as we get that from context.